src/server.rs

```diff
@@ -371,6 +372,35 @@ async fn index(
     } else if let Some((function, route_params)) = router.get_route(req.method(), req.uri().path())
     {
         request.params = route_params;
+
+        // Perform dependency validation
+        // May bottleneck due to calling GIL
+        if function.validate_params {
+            let mut cont = false;
+            Python::with_gil(|py| {
+                let request_hashmap = request.to_hashmap(py).unwrap();
+
+                // Retrieve the FunctionInfo from the route
+                let handler = function.handler.as_ref(py);
+
+                // Import dependency extraction
+                let robyn = py.import("robyn").unwrap();
+
+                // Extract the route's dependencies from the FunctionInfo
+                let dependencies = robyn.call_method1("get_signature", (handler,)).unwrap();
+
+                // Check if the request params match with the dependencies
+                let check_dependencies = robyn.call_method1("check_params_dependencies", (dependencies, request_hashmap,)).unwrap();
+            
+                cont = check_dependencies.extract().unwrap();
```

Comment:
This kinda defeats the purpose of writing Robyn in Rust if we are importing a custom defined function from Python function. Have you considered writing `get_signature` and `check_params_dependenencies` in rust?

Reply:
I've considered perhaps writing both functions in Rust, and if prompted to, I'd probably find a way to do it.
However, it's a little hard to do so since I have to consider working with ForwardRefs in get_signature. The reason I didn't do so for check_params_dependencies is because I have plans to use msgspec not only for json encoding but also for schema validation, since the two features work hand in hand with each other, and it is also possible to define custom classes that represent JSON nesting that do not need to extend from any msgspec's classes, so the user doesn't actually need to learn how to use msgspec (while FastAPI requires the user to define Pydantic models).

Reply:
However, I realize that like in other sections of the code, I could simply import the libs in Rust and write the function in Rust where I'd just call the specific Python functions using the call_method1. However, I'm not sure if this is what you meant. Either way, I'll try my best.

Reply:
@kliwongan , this is exactly what I meant. As creating a binding for a library and then reimporting them feels like circular imports, I am not sure if they will cause any issues or not but this way allows us to keep the code style homogenous throughout. 

> However, I realize that like in other sections of the code, I could simply import the libs in Rust and write the function in Rust where I'd just call the specific Python functions using the call_method1.